polaris-core/src/main/java/org/apache/polaris/core/storage/cache/StorageCredentialCacheKey.java

```diff
@@ -95,16 +88,13 @@ public Set<String> getAllowedWriteLocations() {
     return allowedWriteLocations;
   }
 
-  public @Nullable PolarisCallContext getCallContext() {
-    return callContext;
-  }
-
   @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     StorageCredentialCacheKey cacheKey = (StorageCredentialCacheKey) o;
-    return catalogId == cacheKey.getCatalogId()
+    return Objects.equals(realmId, cacheKey.getRealmId())
```

Comment:
nit: maybe just convert this class to `@PolarisImmutable`?

Cf. `StsDestination`